api/src/main/java/org/openmrs/module/bedmanagement/service/impl/BedManagementServiceImpl.java

```diff
@@ -98,6 +98,22 @@ public AdmissionLocation setBedLayoutForAdmissionLocation(AdmissionLocation admi
 		return admissionLocation;
 	}
 	
+	@Override
+	public void removeBedLayoutOfAdmissionLocationOutSideRowColumn(AdmissionLocation admissionLocation, Integer row,
+	        Integer column) {
+		List<BedLocationMapping> bedLocationMappings = bedManagementDao
+		        .getBedLocationMappingByLocation(admissionLocation.getWard());
+		for (BedLocationMapping bedlocationMapping : bedLocationMappings) {
+			if (bedlocationMapping.getRow() > row || bedlocationMapping.getColumn() > column) {
+				if (bedlocationMapping.getBed() == null) {
+					bedManagementDao.deleteBedLocationMapping(bedlocationMapping);
+				} else {
+					throw new IllegalPropertyException("Invalid row & column");
```

Comment:
This is not the correct error message. It should be something along the lines of
```java
"Cannot downsize bed layout with existing bed in the way at row " + bedlocationMapping.getRow() + "and column " + bedlocationMapping.getColumn()
```
Also check what is happening downstream. This error message is not handled properly at the resource level and is not displayed to the user on the front-end.
You could even i18n this message using two aliases in the message (one for the row number, one for the column number).

Reply:
Also it would have been nice to have `setBedLayoutForAdmissionLocation(..)` able to handle this. So it could be used to both set a new bed layout to an admission location as well as resize an existing bed layout for an admission location.